gammapy/utils/nddata.py

```diff
@@ -0,0 +1,441 @@
+        elif method == 'nearest':
+            return self._eval_linear(values, method='nearest') * self.data.unit
+        else:
+            raise ValueError('Interpolator {} not available'.format(method))
+
+    def add_linear_interpolator(self, **kwargs):
+        """Add `~scipy.interpolate.RegularGridInterpolator`
+
+        The interpolation behaviour of an individual axis can be adjusted by
+        setting the ``log_interpolation`` attribute to True. After that the
+        interpolator has to be setup anew. TODO: Make this somehow more
+        convenient?
+
+        http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.interpolate.RegularGridInterpolator.html
+        """
+        kwargs.setdefault('bounds_error', False)
+
+        from scipy.interpolate import RegularGridInterpolator
+
+        points = [a._interp_nodes() for a in self.axes]
+        values = self.data.value
+
+        self._lininterp = RegularGridInterpolator(points, values, **kwargs)
+
+    def _eval_linear(self, values, method='linear'):
+        """Evaluate linear interpolator
+
+        Input: list of values to evaluate, in correct units and correct order.
+        """
+        if self._lininterp is None:
+            raise ValueError('Linear interpolation requested but no linear '
+                             'interpolator initialized')
+
+        shapes = [np.shape(_)[0] for _ in values]
+        points = list(itertools.product(*values))
+        res = self._lininterp(points, method=method)
+        res = np.reshape(res, shapes)
+
+        return res
+
+    def plot_image(self, ax=None, plot_kwargs = {}, **kwargs):
+        """Plot image
+
+        Only avalable for 2D (after slicing)
+        """
+        import matplotlib.pyplot as plt
+        ax = plt.gca() if ax is None else ax
+
+        data = self.evaluate(**kwargs)
+        if len(data.squeeze().shape) != 2:
+            raise ValueError('Data has shape {} after slicing. '
+                             'Need 2d data for image plot'.format(data.shape))
+
+        ax.set_xlabel('{} [{}]'.format(self.axes[0].name, self.axes[0].unit))
+        ax.set_ylabel('{} [{}]'.format(self.axes[1].name, self.axes[1].unit))
+        ax.imshow(data.transpose(), origin='lower', **plot_kwargs)
+
+    def plot_profile(self, axis, ax=None, **kwargs):
+        """Show data as function of one axis
+
+        Parameters
+        ----------
+        axis : DataAxis
+            data axis to use
+        """
+
+        raise NotImplementedError
+
+        import matplotlib.pyplot as plt
+        ax = plt.gca() if ax is None else ax
+
+        ax_ind = self.get_axis_index(axis)
+        kwargs.setdefault(axis, self.axes[ax_ind])
+
+        x = kwargs.pop(axis)
+
+        y = self.evaluate(**kwargs)
+
+
+class DataAxis(Quantity):
+    """Data axis for unbinned values"""
+    def __new__(cls, energy, unit=None, dtype=None, copy=True, name=None):
```

Comment:
Rename `energy` to something more generic?
Actually I'm not sure what it is ... the docstring says "Data axis for unbinned values"!?

Probably an
```
Examples
-------
```
section with one or two examples would help more than a text description.